From efd58b185c885d72a5fc0264d1a2dbff47080244 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Sun, 5 Aug 2018 15:24:40 +0300
Subject: [PATCH 0533/1051] net: mvpp2: fix short pool frame size

PROBLEM: short pool packet size is ~0 and therefore never used
(short packets are going over long pool).

REQUIREMENT: packets up to 512 bytes should go over short-pool.

FINDING: the MVPP2_BM_SHORT_FRAME_SIZE=512 but should be
 512+NET_SKB_PAD+MVPP2_SKB_SHINFO_SIZE = 512+128+384

FIX:
 Set correct value 1024

Change-Id: Iebe5c944b9dd307c9450f82773a25d536ba746b6
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/59462
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: Stefan Chulski <stefanc@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2.h b/drivers/net/ethernet/marvell/mvpp2/mvpp2.h
index 976acd53bc04..d8da6432f10f 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2.h
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2.h
@@ -655,7 +655,7 @@ enum mvpp2_prs_l3_cast {
 #define MVPP2_BM_COOKIE_POOL_OFFS	8
 #define MVPP2_BM_COOKIE_CPU_OFFS	24
 
-#define MVPP2_BM_SHORT_FRAME_SIZE		512
+#define MVPP2_BM_SHORT_FRAME_SIZE		1024
 #define MVPP2_BM_LONG_FRAME_SIZE		2048
 #define MVPP2_BM_JUMBO_FRAME_SIZE		10240
 /* BM short pool packet size
-- 
2.17.1

